File: backend/app/schemas/department.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class DepartmentCreate(BaseModel):
    name: str = Field(min_length=2, max_length=120)
    description: str | None = None


class DepartmentUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=2, max_length=120)
    description: str | None = None


class DepartmentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: str | None
    user_count: int = 0
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/department.ts
export type Department = {
  id: number;
  name: string;
  description: string | null;
  user_count: number;
  created_at: string;
  updated_at: string;
};

export type DepartmentCreateInput = {
  name: string;
  description: string | null;
};

export type DepartmentUpdateInput = Partial<DepartmentCreateInput>;
